packages/backend/src/open_payments/receiver/model.ts

```diff
@@ -106,9 +120,9 @@ export class Receiver extends ConnectionBase {
   }
 
   public get receivedAmount(): Amount | undefined {
-    if (this.receivedAmountValue !== undefined) {
+    if (this.incomingPayment?.receivedAmount) {
       return {
-        value: this.receivedAmountValue,
+        value: this.incomingPayment.receivedAmount.value,
         assetCode: this.assetCode,
         assetScale: this.assetScale
```

Comment:
Should I add validation in this class to make sure `assetCode` and `assetScale` are the same across `receivedAmount` and `incomingAmount`?